metis-core/metis-core-service/src/test/java/eu/europeana/metis/core/service/TestRedirection.java

```diff
@@ -0,0 +1,269 @@
+                .thenReturn(indexToPublishPluginWithExecutionId);
+
+        when(dataEvolutionUtils.getRootAncestor(any()))
+                .thenReturn(httpHarvestPluginWithExecutionId)
+                .thenReturn(httpHarvestPluginWithExecutionId2);
+
+        final String datasetId = "datasetId";
+        final ObjectId objectId = new ObjectId();
+        final Workflow workflow = getWorkflow(datasetId, objectId, getIndexToPublishPluginMetadata(indexToPreviewPlugin));
+        final Dataset dataset = getTestDataset(datasetId);
+        final PluginWithExecutionId<ExecutablePlugin> predecessor = new PluginWithExecutionId<>("executionId", indexToPreviewPlugin);
+
+        final WorkflowExecution workflowExecution = workflowExecutionFactory.createWorkflowExecution(workflow, dataset, predecessor, priority);
+
+        final AbstractMetisPlugin<IndexToPublishPluginMetadata> abstractMetisPlugin = workflowExecution.getMetisPlugins().stream().findFirst().get();
+        assertTrue(abstractMetisPlugin.getPluginMetadata().isPerformRedirects());
+    }
+
+    @Test
+    void redirectionReviewWithPerformRedirectsWhenRedirectIdsPresent() throws BadContentException {
+        final int priority = 0;
+        final HTTPHarvestPluginMetadata httpHarvestPluginMetadata = getHttpHarvestPluginMetadata();
+        final HTTPHarvestPlugin httpHarvestPlugin = getHttpHarvestPlugin(httpHarvestPluginMetadata,
+                Date.from(Instant.now().minus(42, ChronoUnit.MINUTES)),
+                Date.from(Instant.now().minus(28, ChronoUnit.MINUTES)),
+                getExecutionProgress());
+        final PluginWithExecutionId<ExecutablePlugin> httpHarvestPluginWithExecutionId =
+                new PluginWithExecutionId<>("executionId", httpHarvestPlugin);
+
+        final IndexToPreviewPluginMetadata indexToPreviewPluginMetadata = getIndexToPreviewPluginMetadata();
+        final IndexToPreviewPlugin indexToPreviewPlugin = getIndexToPreviewPlugin(indexToPreviewPluginMetadata, getExecutionProgress());
+
+        final IndexToPublishPluginMetadata indexToPublishPluginMetadata = getIndexToPublishPluginMetadata(indexToPreviewPlugin);
+        final IndexToPublishPlugin indexToPublishPlugin = getIndexToPublishPlugin(indexToPublishPluginMetadata, getExecutionProgress());
+
+        final PluginWithExecutionId<ExecutablePlugin> indexToPublishPluginPluginWithExecutionId =
+                new PluginWithExecutionId<>("executionId", indexToPublishPlugin);
+
+        when(workflowExecutionDao.getLatestSuccessfulExecutablePlugin(anyString(), any(), anyBoolean()))
+                .thenReturn(indexToPublishPluginPluginWithExecutionId);
+        final PluginWithExecutionId<ExecutablePlugin> indexToPreviewPluginPluginWithExecutionId =
+                new PluginWithExecutionId<>("executionId", indexToPreviewPlugin);
+        lenient().when(dataEvolutionUtils.getRootAncestor(any()))
```

Comment:
Although I commented on this before and after reading the mockito documentation about the more strict control over tests, I now think that it is better how you had it before. 
I mean without the stubbing since it is not necessary and makes cleaner tests.
What do you think?